crypto.h

```diff
@@ -115,3 +162,30 @@ UCHAR OvpnCryptoOpcodeExtract(UCHAR op)
 {
     return op >> OVPN_OPCODE_SHIFT;
 }
+
+static inline
+BOOLEAN
+OvpnCryptoAeadUsageLimitReached(UINT64 limit, UINT64 plaintextBlocks, UINT64 highestPid)
+{
+    /* This is the  q + s <=  p^(1/2) * 2^(129/2) - 1 calculation where
+     * q is the number of protected messages (highest_pid)
+     * s Total plaintext length in all messages (in blocks) */
+    return ((limit > 0) && (plaintextBlocks + highestPid) > limit);
+}
+
+static inline
+UINT64
+OvpnCryptoAeadUsageLimit(OVPN_CIPHER_ALG alg)
+{
+    switch (alg)
+    {
+    case OVPN_CIPHER_ALG_NONE:
+        return 0;
+
+    case OVPN_CIPHER_ALG_CHACHA20_POLY1305:
+        return 0;
+
+    default:
+        return (1ull << 36) - 1;
```

Comment:
Maybe add a comment here that this is the limit for AES-GCM

Reply:
Done!